pkg/ottl/ottlfuncs/README.md

```diff
@@ -126,6 +127,36 @@ Examples:
 
 - `IsMatch("string", ".*ring")`
 
+### ParseJSON
+
+`ParseJSON(target)`
+
+The `ParseJSON` factory function returns a `pcommon.Map` struct that is a result of parsing the target string as JSON
+
+`target` is a Getter that returns a string. This string should be in json format.
+
+Unmarshalling is done using [jsoniter](https://github.com/json-iterator/go).
+Each JSON type is converted into a `pdata.Value` using the following map:
+
+```
+JSON boolean -> bool
+JSON number  -> float64
+JSON string  -> string
+JSON null    -> nil
+JSON arrays  -> pdata.SliceValue
+JSON objects -> map[string]any
+```
+
+Examples:
+
+- `ParseToMap("{\"attr\":true}")`
+
+
+- `ParseToMap(attributes["kubernetes"])`
```

Comment:
Kent and I were talking about the resulting key names and in a situation like this it could be nice to prefix all the root keys with `kubernetes` so that you know they came from the kubernetes attribute.  It is efficient to do this during parsing, but I don't like that it requires and extra parameter in the function that other use cases won't need.

We could also handle something like this in a separate Factory Function that allows modification of attribute keys.

@djaglowski does stanza have any precedence for prefixing key names?

Reply:
It does not have the feature. 

I think it could be a useful capability, but wouldn't it be better as an independent function? In general I would encourage us to have simpler functions that can be composed as desired.

Reply:
Separate function would provide the most composability, doing it here would only be about performance.  I am ok leaving this out for now and revisiting if we start seeing depredated performance in our factory function compositions.